File: cpp/Tiket.cpp
#include <bits/stdc++.h> //dia pake libary yang dibutuhin jadi gausah include satu satu
using namespace std;     // ini biar singkat gausah ngetik std::cout << a std::endl;

//deklrarasi
class Tiket{
    private:
        string id_tiket;
        string nama_tiket;
        int jumlah_tiket;
        double harga_tiket;
    public:
        Tiket(){}
        Tiket(string id, string nama, int jumlah, double harga){
            SetID(id);
            SetNama(nama);
            SetJumlah(jumlah);
            SetHarga(harga);
        }

        //setter id
        //const = data konstan jadi bisa buat baca doang
        // & buat referense
        void SetID(const string& id){
            this->id_tiket = id;    //set id
        }
        //setter nama
        void SetNama(const string& nama){
            this->nama_tiket = nama;    //set nama
        }
        //setter jumlah 
        void SetJumlah(const int& jumlah){
            this->jumlah_tiket = jumlah;
        }
        //setter harga
        void SetHarga(const double& harga){
            this->harga_tiket = harga;
        }

        //getter
        // const buar  readonly
        string GetID() const{
            return id_tiket;
        }
        //getter nama
        string GetName() const{
            return nama_tiket;
        }
        string GetNama() const{
            return nama_tiket;
        }
        //getter jumlah
        int GetJumlah() const{
            return jumlah_tiket;
        }
        //getter harga
        double GetHarga() const{
            return harga_tiket;
        }

        //menampilkan data tiket
        void ShowData() const{
            cout << "------------------------" << endl;
            cout << "ID Tiket   : " << id_tiket << endl;
            cout << "Nama Tiket : " << nama_tiket << endl;
            cout << "Jumlah     : " << jumlah_tiket << endl;
            cout << "Harga      : " << fixed << setprecision(0) << harga_tiket << endl;
        }

        //destruktor
        ~Tiket(){}
};

File: python/Tiket.py
class Tiket:
    
    def __init__(self, id_tiket: str, nama_tiket: str, jumlah_tiket: int, harga_tiket: float):
        # inisiasi
        self.__id_tiket = str(id_tiket)
        self.__nama_tiket = str(nama_tiket)
        self.__jumlah_tiket = int(jumlah_tiket)
        self.__harga_tiket = float(harga_tiket)
        
    # getter
    def GetId(self):
        return self.__id_tiket
    
    def GetNama(self):
        return self.__nama_tiket
    
    def GetJumlah(self):
        return self.__jumlah_tiket
    
    def GetHarga(self):
        return self.__harga_tiket
    
    # getter alias (camelCase)
    getId = GetId
    getNama = GetNama
    getJumlah = GetJumlah
    getHarga = GetHarga
    
    # setter
    def SetId(self, id_tiket):
        self.__id_tiket = str(id_tiket)
        
    def SetNama(self, nama_tiket):
        self.__nama_tiket = str(nama_tiket)
    
    def SetJumlah(self, jumlah_tiket):
        self.__jumlah_tiket = int(jumlah_tiket)
            
    def SetHarga(self, harga_tiket):
        self.__harga_tiket = float(harga_tiket)

    # setter alias (camelCase)
    setId = SetId
    setNama = SetNama
    setJumlah = SetJumlah
    setHarga = SetHarga
            
    # metod show data asek
    def ShowData(self):
        print("------------------------")
        print("ID Tiket   : ", self.GetId())
        print("Nama Tiket : ", self.GetNama())
        print("Jumlah     : ", self.GetJumlah())
        print(f"Harga      : {self.GetHarga():.0f}")

    showData = ShowData